libcudacxx/include/cuda/std/__atomic/api/owned.h

```diff
@@ -0,0 +1,134 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __LIBCUDACXX___ATOMIC_API_OWNED_H
+#define __LIBCUDACXX___ATOMIC_API_OWNED_H
+
+#include <cuda/std/detail/__config>
+
+#include <cuda/std/__atomic/api/common.h>
+#include <cuda/std/__atomic/order.h>
+#include <cuda/std/__atomic/scopes.h>
+#include <cuda/std/__atomic/types.h>
+#include <cuda/std/__atomic/wait/notify_wait.h>
+#include <cuda/std/__atomic/wait/polling.h>
+#include <cuda/std/__type_traits/conditional.h>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+template <typename _Tp, typename _Sco>
+struct __atomic_common
+{
+  _CCCL_HOST_DEVICE constexpr inline __atomic_common(_Tp __v)
+      : __a(__v)
+  {}
+
+  _CCCL_HOST_DEVICE constexpr inline __atomic_common()
+      : __a()
+  {}
+
+  __atomic_storage_t<_Tp> __a;
+
+#if defined(_LIBCUDACXX_ATOMIC_ALWAYS_LOCK_FREE)
```

Comment:
Yeah, probably, we might want to check if we're on a system that doesn't have 8-byte atomics.

It might be possible in the future that we want to enable 128b depending on what we can detect.

I'm in favor of leaving a customization point.